servers/src/grin/seed.rs

```diff
@@ -254,11 +259,11 @@ fn connect_to_seeds_and_preferred_peers(
 ) {
 	// check if we have some peers in db
 	// look for peers that are able to give us other peers (via PEER_LIST capability)
-	let peers = peers.find_peers(p2p::State::Healthy, p2p::Capabilities::PEER_LIST, 100);
+	let mut peers = peers.find_peers(p2p::State::Healthy, p2p::Capabilities::PEER_LIST, 100);
 
 	// if so, get their addresses, otherwise use our seeds
 	let mut peer_addrs = if peers.len() > 3 {
-		peers.iter().map(|p| p.addr).collect::<Vec<_>>()
+		peers.drain(0..).map(|p| p.addr).collect::<Vec<_>>()
```

Comment:
Why `drain` is preferable here?

Reply:
Not sure, this is from Igno's initial work. Can change if `iter()` is better/more idiomatic.